bpf/bpf_overlay.c

```diff
@@ -310,6 +310,41 @@ int tail_handle_ipv4(struct __ctx_buff *ctx)
 }
 #endif /* ENABLE_IPV4 */
 
+#ifdef ENABLE_IPSEC
+static __always_inline bool is_esp(struct __ctx_buff *ctx, __u16 proto)
+{
+	void *data, *data_end;
+	__u8 protocol = 0;
+#ifdef ENABLE_IPV6
+	struct ipv6hdr *ip6;
+#endif
+#ifdef ENABLE_IPV4
+	struct iphdr *ip4;
```

Comment:
We usually simply add `__maybe_unused` instead of macros for the variables. Otherwise it quickly becomes very heavy in macros :smile_cat: